.github/workflows/tests.yml

```diff
@@ -405,11 +417,30 @@ jobs:
         if: ${{ !cancelled() && github.event_name != 'schedule' }}
         run: |
           if [[ "${{ runner.os }}" != 'macOS' ]]; then
-            lcov --rc lcov_branch_coverage=1 --capture --directory . \
-              --output-file coverage.info
-            lcov --rc lcov_branch_coverage=1 --output-file coverage.info \
-              --extract coverage.info $PWD/src/'*' $PWD/lib/'*'
-            lcov --rc lcov_branch_coverage=1 --list coverage.info
+            if [[ "${{ matrix.os }}" != ubuntu-24.04 ]]; then
+              lcov --rc lcov_branch_coverage=1 \
+                --capture \
+                --directory . \
+                --output-file coverage.info
+              lcov --rc lcov_branch_coverage=1 \
+                --output-file coverage.info \
+                --extract coverage.info $PWD/src/'*' $PWD/lib/'*'
+              lcov --rc lcov_branch_coverage=1 \
+                --list coverage.info
+            else  # ubuntu-24.04
+              lcov --branch-coverage \
+                --ignore-errors 'mismatch,unused' \
+                --capture \
+                --directory . \
+                --output-file coverage.info
+              lcov --branch-coverage \
+                --ignore-errors 'mismatch,unused' \
+                --output-file coverage.info \
+                --extract coverage.info $PWD/src/'*' $PWD/lib/'*'
+              lcov --branch-coverage \
+                --ignore-errors 'mismatch,unused' \
+                --list coverage.info
+            fi
```

Comment:
AFAICS the only difference for ubuntu 24.04 is adding `--ignore-errors`. If that's true, it may me easier to follow by definining a variable $IGNORE_ERRORS_ARG, which is empty or not, and just always do ` lcov --branch-coverage $IGNORE_ERRORS_ARG ...`

Reply:
That's true-enough that it probably makes sense.  The other difference is that the deprecated `--rc lcov_branch_coverage=1` is replaced with the functionally equivalent `--branch-coverage`.

The former remains supported, and it'd be straightforward to add an explanatory comment alongside an environment variable like that.

One nitpick/suggestion: would `LCOV_IGNORE_ERRORS` seem reasonable for the env var name instead?